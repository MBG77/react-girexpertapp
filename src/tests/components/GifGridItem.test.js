import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {

    const title = 'javascript';
    const url = 'https://media0.giphy.com/media/xT9IgzoKnwFNmISR8I/giphy.gif?cid=fabaeeb8ziulp54a25j9ktiihtlgs86gybo5xcyurg5l9mm6&rid=giphy.gif';
    let wrapper = shallow(<GifGridItem title={ title } url={ url }/>);

    beforeEach(() => {
        wrapper = shallow(<GifGridItem title={ title } url={ url }/>);
    })
    
    test('debe mostrar el componente <GifGridItem /> correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de tener un pÃ¡rrafo con el title', () => {

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
        
    })

    test('debe tener la imagen igual al url y alt de los props.', () => {
        const img = wrapper.find('img');
        // console.log(img.prop('src'));
        expect( img.prop('src') ).toBe(url)
        expect( img.prop('alt') ).toBe(title);
    })

    test('debe tener animate__fadeIn', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        
        expect( className.includes('animate__fadeIn') ).toBe( true );
    })
    
    
    
    
})
