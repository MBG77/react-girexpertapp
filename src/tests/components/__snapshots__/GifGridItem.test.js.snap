// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el componente <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="javascript"
    src="https://media0.giphy.com/media/xT9IgzoKnwFNmISR8I/giphy.gif?cid=fabaeeb8ziulp54a25j9ktiihtlgs86gybo5xcyurg5l9mm6&rid=giphy.gif"
  />
  <p>
    javascript
  </p>
</div>
`;
